backend-rust/src/graphql_api/baker_metrics.sql

```diff
@@ -1,8 +1,29 @@
 SELECT
     bucket_time.bucket_start as "bucket_time!",
-    COALESCE(before_bucket.total_bakers_added, 0) - COALESCE(before_bucket.total_bakers_removed, 0) as "bucket_previous_count_total!",
-    COALESCE(after_bucket.total_bakers_added, 0) - COALESCE(before_bucket.total_bakers_added, 0) as "bucket_bakers_added!",
-    COALESCE(after_bucket.total_bakers_removed, 0) - COALESCE(before_bucket.total_bakers_removed, 0) as "bucket_bakers_removed!"
+    COALESCE((SELECT
+        total_bakers_added
+     FROM metrics_bakers
+     WHERE block_height < before_bucket.height
+     ORDER BY block_height DESC
+     LIMIT 1), 0) as "added_before!",
+    COALESCE((SELECT
+        total_bakers_removed
+     FROM metrics_bakers
+     WHERE block_height < before_bucket.height
+     ORDER BY block_height DESC
+     LIMIT 1), 0) as "removed_before!",
+    COALESCE((SELECT
+        total_bakers_added
+     FROM metrics_bakers
+     WHERE block_height < after_bucket.height
+     ORDER BY block_height DESC
+     LIMIT 1), 0) as "added_after!",
+    COALESCE((SELECT
+        total_bakers_removed
+     FROM metrics_bakers
+     WHERE block_height < after_bucket.height
+     ORDER BY block_height DESC
+     LIMIT 1), 0) as "removed_after!"
```

Comment:
Some further optimization could be to have only two sub-queries instead of the four ones (meaning one query for the `after_bucket` block and one for the `before_bucket` block. Each query would query two rows `total_bakers_removed/total_bakers_removed`.

Reply:
I tried but the subqueries may only return one column in this scenario.